### PR TITLE
Language reference

### PR DESCRIPTION
**Pull request by UnkindPartition**

### CODE REVIEW

File: reference/supported_types.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+* ``Double``
+* ``Boolean``
+* ``Text``
+
+.. ByteArray
+
+More complex types can be built out of primitive ones:
+
+* Tuples: ``(type1, type2, ...)``. A tuple must have at least two fields.
+* Structs: ``{ name1: type1, name2: type2, ... }``
```
Comment (rjmk) [inline reference/supported_types.rst]: Any reason not to include 
```
Sums: ``Name1 type1a ... | Name2 type2a ... | ...``
```
?

File: reference/workflow_language.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  ``A.y`` must be a function that produces lists of values,
+  in which case ``concat A.y`` is a function that "unpacks" the lists
+  and yields the same values one by one.
+
+* Filter: ``filter A.y`` or ``A.y?``
+
+  ``A.y`` must be a function that produces "optional" (potentially missing) values,
+  in which case ``filter A.y`` is a function that filters out missing values.
+
+* Type application. Some functions (notably, sources and sinks) can be specialized
```
Comment (rjmk) [inline reference/workflow_language.rst]: "notably, some sources and sinks" / "notably, the built-in sources and sinks"? I don't think sources and sinks in general have to be polymorphic (The websocket ones aren't, right?)

File: reference/workflow_language.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  Parameters are used to configure sources and sinks —
+  for instance, to specify how to connect to a PostgreSQL database.
+
+  Parameters can also be used to configure user-defined modules.
+  Inside a Python nstack method, the value of parameter ``foo`` can be accessed as
+  ``self.args["foo"]``.
+
+The workflow language supports line and block comments.
+Line comments start with ``--`` and extend until the end of line.
```
Comment (rjmk) [inline reference/workflow_language.rst]: I think we're `//` not `--`

File: reference/supported_types.rst (hunk 1 above)
Comment (UnkindPartition) [reply reference/supported_types.rst]: No reason — I totally missed it. Thanks!

File: reference/workflow_language.rst (hunk 3 above)
Comment (UnkindPartition) [reply reference/workflow_language.rst]: You're right — apparently Mandeep silently changed the comment syntax :)

File: reference/workflow_language.rst (hunk 2 above)
Comment (UnkindPartition) [reply reference/workflow_language.rst]: But "the built-in sources and sinks" is imprecise in exactly the same way, no?

Adding "some" works, though.

File: reference/workflow_language.rst (hunk 3 above)
Comment (mands) [reply reference/workflow_language.rst]: foiled by `git blame` :) sorry about that!

**Review by rjmk — APPROVED** [review]
Awesome!

**Pull request merged**